homeassistant/components/cast/media_player.py

```diff
@@ -505,6 +722,8 @@ def async_cast_removed(discover: ChromecastInfo):
         self.async_schedule_update_ha_state()
 
         await self.hass.async_add_job(self._chromecast.disconnect)
+        if self._dynamic_group_cast is not None:
+            await self.hass.async_add_job(self._dynamic_group_cast.disconnect)
```

Comment:
We should not use `add_job`. If it has to be run in the executor, use `async_add_executor_job`

Reply:
OK, but then the existing code which is also using add_job should be modified too?

Reply:
let's not add too much to this already big PR. So only if it's really close, like the line above this one.

Reply:
Can that be cleaned up all at once in a separate PR instead?